rust/agents/relayer/src/msg/metadata/aggregation.rs

```diff
@@ -140,21 +147,34 @@ impl MetadataBuilder for AggregationIsmMetadataBuilder {
                 .map(|ism_address| self.base.build_ism(*ism_address)),
         )
         .await;
-
-        let filtered_sub_module_metas = metas
+        let (valid_sub_modules, invalid_sub_modules): (Vec<_>, Vec<_>) = metas
             .into_iter()
             .enumerate()
             .zip(sub_modules.into_iter())
-            .filter_map(|((index, meta_result), sub_module_result)| {
-                match (meta_result, sub_module_result) {
-                    (Ok(Some(meta)), Ok(ism)) => Some(IsmAndMetadata::new(ism, index, meta)),
-                    _ => None,
-                }
+            .partition(|&((_, ref meta_result), ref sub_module_result)| {
+                matches!((meta_result, sub_module_result), (Ok(Some(_)), Ok(_)))
+            });
+        let valid_sub_module_metas: Vec<_> = valid_sub_modules
+            .into_iter()
+            .map(|((index, meta_result), sub_module_result)| {
+                IsmAndMetadata::new(
+                    sub_module_result.unwrap(),
+                    index,
+                    meta_result.unwrap().unwrap(),
+                )
             })
             .collect();
 
+        let mut invalid_isms: Vec<(ModuleType, H256)> = Vec::new();
+        // get the address and module_type for each invalid sub module (for logging in cheapest_valid_metas)
+        for entry in &invalid_sub_modules {
+            let (_, ism_result) = entry;
+            if let Ok(ism_box) = ism_result {
+                invalid_isms.push((ism_box.module_type().await.unwrap(), ism_box.address()));
```

Comment:
We'd end up doing a lot of repeated queries with this unless we cache. 

I'd be happy to just print the ism addresses for now, that's still an improvement:
```rust
let invalid_isms: Vec<H256> = invalid_sub_modules.into_iter().map(|(_, addr)| addr).collect();
```

Also since `invalid_isms` isn't relevant to `cheapest_valid_metas`, you could take the `info!("Could not fetch all metadata: ...)` out of there and add it after this call if `maybe_aggregation_metadata.len() < threshold`